pyo3-macros-backend/src/quotes.rs

```diff
@@ -0,0 +1,15 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+
+pub(crate) fn ok_wrap(obj: TokenStream) -> TokenStream {
+    quote! {
+        _pyo3::impl_::pymethods::OkWrap::wrap(#obj, py)
+            .map_err(::core::convert::Into::into)
+    }
+}
+
+pub(crate) fn map_result_into_ptr(result: TokenStream) -> TokenStream {
+    quote! {
+        #result.map(_pyo3::IntoPyPointer::into_ptr)
+    }
+}
```

Comment:
I think this function can be removed (and code moved out of the macros) by moving the `Py<PyAny> -> *mut ffi::PyObject` conversion into the trampolines. That'll also have a positive effect on code size, so I'll enact that as a follow-up within a few days.